subsystems/intakesubsystem.py

```diff
@@ -0,0 +1,140 @@
+from util.angleoptimize import intakeAccountForSillyEncoder
+
+import constants
+
+
+class IntakeSubsystem(Subsystem):
+    class IntakeState(Enum):
+        Intaking = auto()
+        Score = auto()
+        Scoring = auto()
+        Knock = auto()
+
+    def __init__(self) -> None:
+        Subsystem.__init__(self)
+        self.setName(__class__.__name__)  # basic subsystem boilerplate
+
+        self.pivotEncoder = CTREEncoder(
+            constants.kPivotEncoderID, constants.kPivotEncoderOffset
+        )
+
+        self.intakeMotor = Talon(
+            constants.kIntakeCANID,
+            constants.kIntakeName,
+            constants.kIntakePGain,
+            constants.kIntakeIGain,
+            constants.kIntakeDGain,
+            constants.kIntakeInverted,
+        )
+        self.intakeMotor.setCurrentLimit(constants.kIntakeCurrentLimit)
+        self.pivotMotor = Talon(
+            constants.kPivotCANID,
+            constants.kPivotName,
+            constants.kPivotPGain,
+            constants.kPivotIGain,
+            constants.kPivotDGain,
+            constants.kPivotInverted,
+            moMagicAccel=constants.kPivotAccel,
+            moMagicVel=constants.kPivotVel,
+        )
+        self.pivotMotor.setNeutralMode(Talon.NeutralMode.Brake)
+        # self.positionDebouncer = Debouncer(0.1, Debouncer.DebounceType.kRising)
+
+        self.resetPivot()
+
+        self.state = self.IntakeState.Score
+        self.targetAngle = Rotation2d()
+
+        self.intakeAtPositionPublisher = (
+            NetworkTableInstance.getDefault()
+            .getBooleanTopic(constants.kIntakeAtPositionKey)
+            .publish()
+        )
+        self.pivotAnglePublisher = (
+            NetworkTableInstance.getDefault()
+            .getFloatTopic(constants.kPivotAngleKey)
+            .publish()
+        )
+        self.intakeStatePublisher = (
+            NetworkTableInstance.getDefault()
+            .getStringTopic(constants.kIntakeStateKey)
+            .publish()
+        )
+
+    def periodic(self) -> None:
+        # a lot simpler when there isn't a convoluted intake sequence
+        if self.state == self.IntakeState.Intaking:
+            self.setPivotAngle(constants.kIntakingAngle)
+            self.intakeMotor.set(
+                Talon.ControlMode.Percent, -1 * constants.kIntakeMotorSpeed
+            )
+        elif self.state == self.IntakeState.Score:
+            self.setPivotAngle(constants.kScoreAngle)
+            self.intakeMotor.set(Talon.ControlMode.Percent, 0)
+        elif self.state == self.IntakeState.Scoring:
+            self.setPivotAngle(constants.kScoreAngle)
+            self.intakeMotor.set(Talon.ControlMode.Percent, constants.kIntakeMotorSpeed)
+        elif self.state == self.IntakeState.Knock:
+            self.setPivotAngle(constants.kKnockAngle)
+            self.intakeMotor.set(Talon.ControlMode.Percent, 0)
+
+        self.intakeAtPositionPublisher.set(self.intakeAtPosition())
```

Comment:
any 3d pose publishing of the intake?

Reply:
will be done on a new branch after elevator and intake are merged